### PR TITLE
WT-12886 Add terse output mode to backup tool.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: tools/backup_analysis.py [hunk 1]
```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
...
 
 #
 # Print a detailed summary of all of the blocks and bytes over all of the files. We accumulated
 # the changes as we went through each of the files.
 #
 def print_summary(opts):
-    print('SUMMARY')
+    print('SUMMARY:')
     # Calculate overall totals from each of the different types first.
     total_bytes = 0
```
Comment (ddanderson) [inline tools/backup_analysis.py:292]: Looks like `total_bytes` is no longer used?

File: tools/backup_analysis.py [hunk 2]
```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
...
+            else:
+                # Print a terse summary all on one line.
+                all_bytes = opts.granularity * ts.gran_blocks
+                pct_bytes = round(abs(ts.bytes / all_bytes * 100))
+                print(
+                    f'{n}: files: {ts.files_changed} of {ts.files};'
+                    f' bytes: {ts.bytes} of {all_bytes} {pct_bytes}%;'
+                    f' blocks: {ts.chg_blocks} of {ts.gran_blocks} {chg_blocks}%;'
+                    f' {pct20_blocks}% <20%;'
+                    f' {pct80_blocks}% 80>%')
```
Comment (ddanderson) [inline tools/backup_analysis.py:335]: `80>%` maybe should be `<80%`?

File: tools/backup_analysis.py [hunk 3]
```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
...
+    pct80_blocks = 0
     if chg_blocks != 0:
         ts.files_changed += 1
         pct20_blocks = round(abs(pct20_count / chg_blocks * 100))
         pct80_blocks = round(abs(pct80_count / chg_blocks * 100))
-        print(f'{filename}: smallest 20%: {pct20_count} of {chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
-        print(f'{filename}: largest 80%: {pct80_count} of {chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
-    print("")
+    # Print the time even if no changes because we may want to know how long it took to not
+    # see any changes.
```
Comment (ddanderson) [inline tools/backup_analysis.py:258]: This comment `#Print the time` should be nested in the `if` clause, where the time is printed.

File: tools/backup_analysis.py [hunk 4]
```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
...
+            print(f'{filename}: smallest 20%: {pct20_count} of {chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
+            print(f'{filename}: largest 80%: {pct80_count} of {chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
+        print("")
+    else:
+        # Print a terse summary all on one line.
+        print(
+            f'{filename}: bytes: {total_bytes_diff} of {min_size} {chg_byte_pct}%;'
+            f' blocks: {chg_blocks} of {total_gran_blocks} {chg_block_pct}%;'
+            f' {pct20_blocks}% <20%;'
+            f' {pct80_blocks}% 80>%')
```
Comment (ddanderson) [inline tools/backup_analysis.py:283]: `80>%` maybe should be `<80%?`

**Review by ddanderson — APPROVED** [review]
LGTM, modulo potential small fixes, take them if they make sense.

File: tools/backup_analysis.py (hunk 2 above)
Comment (ddanderson) [reply tools/backup_analysis.py]: I still think it's a bit weird though the +/- makes it less weird.

Here's why I'm confused.  You said:
`what it means is percent of blocks that had less than 20% of the bytes in the block changed`.
Translating the "less than" in that sentence it looks like : `%blocks that had <20%`

whereas: `versus blocks that had more than 80% of the bytes in the block changed`
for me translates as: `%blocks that had >80%`.

Maybe you have a different way to visualize the symbols...?
Ultimately, you're the primary customer of this tool, so whatever makes sense to you is fine with me!

File: tools/backup_analysis.py (hunk 3 above)
Comment (sueloverso) [reply tools/backup_analysis.py]: Fixed

File: tools/backup_analysis.py (hunk 1 above)
Comment (sueloverso) [reply tools/backup_analysis.py]: Fixed

File: tools/backup_analysis.py (hunk 2 above)
Comment (sueloverso) [reply tools/backup_analysis.py]: I did move the "title" before the percentage to be consistent with the other "titles" in the line FWIW. I went with the "+/-".

Comment (ddanderson) [reply tools/backup_analysis.py]: Yes, that looks better to me!  Thanks.

**Pull request merged**